be/src/storage/lake/compaction_policy.h

```diff
@@ -41,4 +41,6 @@ double cumulative_compaction_score(const TabletMetadataPB& metadata);
 
 double base_compaction_score(const TabletMetadataPB& metadata);
 
+double primary_compaction_score(const TabletMetadataPB& metadata);
```

Comment:
Perhaps we should only expose an interface for calculating compaction scores:
compaction_policy.h
```
double compaction_score(const TabletMetadataPB& metadata);
``` 

compaction_policy.cpp
```
double compaction_score(const TabletMetadataPB& metadata) {
    if (metadata.keys_type() == PRIMARY_KEYS) {
        return primary_compaction_score(metadata);
    } else {
        return std::max(cumulative_compaction_score(metadata), base_compaction_score(metadata));
    }
}
```